drizzlepac/hlautils/astrometric_utils.py

```diff
@@ -292,6 +292,88 @@ def find_gsc_offset(image, input_catalog='GSC1', output_catalog='GAIA'):
+        Image array (ndarray object) with sources to be identified
+    fwhm : float
+        Value of FWHM to use for creating a Gaussian2DKernel object in case no suitable source
+        can be identified in the image.
+    threshold : float
+        Value from the image which serves as the limit for determining sources.
+        If None, compute a default value of (background+5*rms(background)).
+        If threshold < 0.0, use absolute value as scaling factor for default value.
+    source_box : int
+        Size of box (in pixels) which defines the minimum size of a valid source.
+    isolation_size : int
+        Separation (in pixels) to use to identify sources that are isolated from any other sources
+        in the image.
+    saturation_limit : float
+        Flux in the image that represents the onset of saturation for a pixel.
+    Notes
+    ------
+    Ideally, it would be best to determine the saturation_limit value from the data itself,
+    perhaps by looking at the pixels flagged (in the DQ array) as saturated and selecting
+    the value less than the minimum flux of all those pixels, or maximum pixel value in the
+    image if non-were flagged as saturated (in the DQ array).
+    """
+
+    # Try to use PSF derived from image as detection kernel
+    # Kernel must be derived from well-isolated sources not near the edge of the image
+    kern_img = imgarr.copy()
+    edge = source_box * 2
+    kern_img[:edge, :] = 0.0
+    kern_img[-edge:, :] = 0.0
+    kern_img[:, :edge] = 0.0
+    kern_img[:, -edge:] = 0.0
+    peaks = photutils.detection.find_peaks(kern_img, threshold=threshold, box_size=isolation_size)
+    if len(peaks['peak_value'][peaks['peak_value'] > saturation_limit]):
+        # Make sure only peaks less than saturation limit are evaluated
+        peaks['peak_value'][peaks['peak_value'] >= saturation_limit] = 0.
+    # Sort based on peak_value to identify brightest sources for use as a kernel
+    peaks.sort('peak_value')
+
+    wht_box = 2 # Weight image cutout box size is 2 x wht_box + 1 pixels on a side
+
+    # Identify position of brightest, non-saturated peak (in numpy index order)
+    for peak_ctr in range(-1, -1 * len(peaks) - 1, -1):
+        kernel_pos = [peaks['y_peak'][peak_ctr], peaks['x_peak'][peak_ctr]]
+
+        kernel = imgarr[kernel_pos[0] - source_box:kernel_pos[0] + source_box + 1,
+                        kernel_pos[1] - source_box:kernel_pos[1] + source_box + 1]
+
+        kernel_wht = whtarr[kernel_pos[0] - wht_box:kernel_pos[0] + wht_box + 1,
+                        kernel_pos[1] - wht_box:kernel_pos[1] + wht_box + 1]
+
+        # search square cut-out (of size 2 x wht_box + 1 pixels on a side) of weight image centered on peak coords for
+        # zero-value pixels. Reject peak if any are found.
+        if len(np.where(kernel_wht == 0.)[0]) == 0:
+            log.info("Kernel source PSF located at [{},{}]".format(kernel_pos[1], kernel_pos[0]))
+            break
+        else:
```

Comment:
The value of 'kernel' is purely local to the loop, so this else statement is probably not necessary at all.

Reply:
issue resolved in commit [052aba3](*https://github.com/spacetelescope/drizzlepac/pull/420/commits/052aba3da7c5e525c14b96d99171a754a4c99900)